tapdb/addrs.go

```diff
@@ -890,6 +924,52 @@ func fetchEvent(ctx context.Context, db AddrBook, eventID int64,
 	}, nil
 }
 
+// fetchEventByOutpoint fetches a single address event identified by its address
+// and outpoint.
+func fetchEventByOutpoint(ctx context.Context, db AddrBook,
+	addr *address.AddrWithKeyInfo, outpoint wire.OutPoint) (*address.Event,
+	error) {
+
+	dbEvent, err := db.FetchAddrEventByAddrKeyAndOutpoint(
```

Comment:
We could return a concrete err to the caller if the query returns `ErrNoRows`, which it should when there's no match since it specifies `:one`.

Reply:
Fair point! Also added this method to the unit tests.